lib/skate/detours/trip_modification.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Skate.Detours.TripModification do
+  @moduledoc """
+  A module that produces GTFS-flavored TripModification structs for a detour
+
+  More info: https://gtfs.org/realtime/reference/#message-tripmodifications
+  """
+
+  alias Schedule.Gtfs.RoutePattern
+  alias Schedule.Gtfs.Stop
+
+  defmodule Input do
+    @moduledoc """
+    Input for `Skate.Detours.TripModification.for/1`.
+    """
+
+    @type t :: %__MODULE__{
+            route_pattern: RoutePattern.t(),
+            missed_stops: [Stop.t()]
+          }
+
+    defstruct [:route_pattern, :missed_stops]
+  end
+
+  defmodule SelectedTrip do
+    @moduledoc """
+    Struct representing the `selected_trips` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-selectedtrips
+    """
+
+    @type t :: %__MODULE__{
+            trip_ids: [String.t()]
+          }
+
+    defstruct [:trip_ids]
+  end
+
+  defmodule Modification do
+    @moduledoc """
+    Struct representing the `modifications` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-modification
+    """
+
+    @type t :: %__MODULE__{
+            start_stop_selector: String.t(),
+            end_stop_selector: String.t()
+          }
+
+    defstruct [:start_stop_selector, :end_stop_selector]
+  end
+
+  @type t :: %__MODULE__{
+          selected_trips: [SelectedTrip.t()],
+          modifications: [Modification.t()]
+        }
+  defstruct [:selected_trips, :modifications]
+
+  @spec for(input :: Input.t()) :: __MODULE__.t()
+  @doc """
+  A function that takes data about a detour and stitches it together to form a GTFS-flavored
+  struct.
+
+
+  ## Examples
+      iex> Skate.Detours.TripModification.for(
+      ...>   %Skate.Detours.TripModification.Input{
+      ...>     route_pattern: build(:gtfs_route_pattern, representative_trip_id: "01-00"),
+      ...>     missed_stops: [
+      ...>       build(:gtfs_stop, id: "ABC123"),
+      ...>       build(:gtfs_stop, id: "ABC124"),
+      ...>       build(:gtfs_stop, id: "ABC129"),
+      ...>     ]
+      ...>   }
+      ...> )
+      %Skate.Detours.TripModification{
+        selected_trips: [
+          %Skate.Detours.TripModification.SelectedTrip{
+            trip_ids: ["01-00"]
+          }
+        ],
+        modifications: [
+          %Skate.Detours.TripModification.Modification{
+            start_stop_selector: "ABC123",
+            end_stop_selector: "ABC129"
+          }
+        ]
+      }
+  """
+  def for(%Input{
+        route_pattern: %RoutePattern{representative_trip_id: trip_id},
+        missed_stops: missed_stops
+      }) do
+    %__MODULE__{
+      selected_trips: [
+        %SelectedTrip{trip_ids: [trip_id]}
+      ],
+      modifications: [
+        %Modification{
+          start_stop_selector: missed_stops |> List.first() |> Map.get(:id),
```

Comment:
In the medium-to-long term, we'll need to use stop sequences here instead of stop ID's, but that'll involve some changes to the missed stops algorithm.

I'm thinking we should probably do that as a near-term follow-up, because otherwise this will sneak back up on us. But I was thinking that this PR didn't need to be blocked on that! 😅

Reply:
> but that'll involve some changes to the missed stops algorithm.
Can you elaborate? My understanding was that it's not missed stops that needs to change, but the underlying `Stop` struct needs to have `stop_sequence`

Reply:
Oh, that would also work.

I had been thinking that the missed stops algorithm should include the index into the array, so instead of returning:
```elixir
[
  %Stop{...},
  %Stop{...}
]
```

It would return
```elixir
[
  {4, %Stop{...}},
  {5, %Stop{...}}
]
```

or something like that.

But I like the idea of putting `stop_sequence` directly into the data structure better - I just didn't realize that was an option!

What do you think... follow-up, or do it as part of this PR?

Reply:
> I had been thinking that the missed stops algorithm should include the index into the array

Just noting this: This also won't work AFAIK, because we need to source `stop_sequence` from GTFS data
> `stop_times.txt`
> `stop_sequence` 	`Non-negative integer` 	`Required`
>
> Order of stops, location groups, or GeoJSON locations for a particular trip. The values must increase along the trip but do not need to be consecutive.
> _Example: The first location on the trip could have a stop_sequence=1, the second location on the trip could have a stop_sequence=23, the third location could have a stop_sequence=40, and so on._
>
> -- https://gtfs.org/schedule/reference/#stop_timestxt

Reply:
> But I like the idea of putting `stop_sequence` directly into the data structure better - I just didn't realize that was an option!
> 
> What do you think... follow-up, or do it as part of this PR?

Definitely follow-up, because this involves parsing the `stop_sequence` data from GTFS and storing it, then changing the way we source stop data to include this when we get it from a trip. Then we need to add the field to `Stop` and make sure it doesn't affect current `stops.txt` parsing code, but is filled when we get stops associated with a trip. (that's just off the top of my head, I may have repeated myself or not said everything, but I figure we'll get there okay)

Reply:
(Just a little side note, because I think it's neat but I don't think I mentioned or wrote down this, but the reason that I originally "architechted" MissedStops to sort objects which implement `Util.Location.From` instead of doing something "easier" or "simplier" at first, is because of this situation with `stop_sequence`!)

Reply:
> What do you think... follow-up, or do it as part of this PR?

A random thought though, to be in compliance with GTFS-TM spec, maybe we should be asserting that the associated trip does not contain duplicate `stop_id`'s?

That'd be great if it could be in this PR, but the original spec of this module is to be a "simple transformer" and ideally doesn't need to access any outside data. So either the caller would need to provide a Route Pattern and it's Shape + Stops for this module to assert on, or the caller would need to assert, which then doesn't centralize the assertion.

Anyway, not blocking, "just food for thought".

Reply:
I like that idea!

I _think_ that would be an attribute of the route pattern, so if I can easily get that into the route pattern here, I'll add it to this PR.

That does remind me that `{:ok, result}` is the Elixir Way™, so it's worth switching to that!

Reply:
Long-term fix: [⚙️GTFS-TM: Send stop sequence instead of stop ID as the stop selectors](https://app.asana.com/0/1206070633318963/1207634607164315/f)

Short-term fix (will do v. soon): [⚙️GTFS-TM: (Stopgap) return an error if a route-pattern has any duplicate stops](https://app.asana.com/0/1206070633318963/1207635746210198/f)